release_notes/ocp-4-5-release-notes.adoc

```diff
@@ -242,6 +242,182 @@ To support this feature, OLM now enforces CRD upgrades are safer by ensuring
+use the Operator API to install an Operator bundle in a cluster without being
+part of a catalog, shortening the route for developers to test their packaging.
+
+To enable this Technology Preview feature:
+
+.Procedure
+
+. Disable xref:../architecture/architecture-installation.adoc#unmanaged-operators_architecture-installation[Cluster Version Operator (CVO) management] of the OLM:
++
+----
+$ oc patch clusterversion version \
+    --type=merge -p \
+    '{"spec": {"overrides":[{"kind": "Deployment", "name": "olm-operator", "namespace": "openshift-operator-lifecycle-manager", "unmanaged": true, "group": "apps"}]}}'
+----
+
+. Add the `OperatorLifecycleManagerV2=true` xref:../nodes/clusters/nodes-cluster-enabling-features.adoc#nodes-cluster-enabling[FeatureGate]
+to the OLM Operator.
+
+.. Edit the OLM Operator's Deployment:
++
+----
+$ oc -n openshift-operator-lifecycle-manager \
+    edit deployment olm-operator
+----
+
+.. Add the following flag to the Deployment's `args` section:
++
+----
+...
+    spec:
+      containers:
+      - args:
+...
+        - --feature-gates
+        - OperatorLifecycleManagerV2=true
+----
+
+.. Save your changes.
+
+. Install an Operator using the normal OperatorHub method if you have not already;
+this example uses an etcd Operator installed in the project `test-project`.
+
+. Create a new Operator resource for the installed etcd Operator.
+
+.. Save the following to a file:
++
+.`etcd-test-op.yaml` file
+----
+apiVersion: operators.coreos.com/v2alpha1
+kind: Operator
+metadata:
+  name: etcd-test
+----
+
+.. Create the resource in `test-project`:
++
+----
+$ oc create -f etcd-test-op.yaml -n test-project
+----
+
+. To have the installed Operator opt in to the new API, apply the
+`operators.coreos.com/etcd-test` label to the following objects related to your
+Operator:
++
+--
+* Subscription
+* InstallPlan
+* ClusterServiceVersion
+* Any CRDs owned by the Operator
+--
++
+[NOTE]
+====
+In a future release, these objects will be automatically labeled for any
+Operators where the CSV was installed using a Subscription.
+====
++
+For example:
++
+----
+$ oc label sub etcd operators.coreos.com/etcd-test="" -n test-project
+$ oc label ip install-6c5mr operators.coreos.com/etcd-test="" -n test-project
+$ oc label csv etcdoperator.v0.9.4 operators.coreos.com/etcd-test="" -n test-project
+$ oc label crd etcdcluster.etcd.database.coreos.com operators.coreos.com/etcd-test="" -n test-project
+$ oc label crd etcdbackups.etcd.database.coreos.com operators.coreos.com/etcd-test="" -n test-project
+$ oc label crd etcdrestores.etcd.database.coreos.com operators.coreos.com/etcd-test="" -n test-project
+----
+
+. Verify your Operator has opted in to the new API.
+
+.. List all `operators` resources:
++
+----
+$ oc get operators --all-namespaces
+
+NAME        AGE
+etcd-test   17m
+----
+
+.. Inspect your Operator's details and note that the objects you labeled are
+represented:
++
+----
+$ oc describe etcd-test -n test-project
```

Comment:
the command should be "oc get operator etcd-test -o yaml"

Reply:
the output is like the following, note: it is Default project, not test-project project
```console
apiVersion: operators.coreos.com/v2alpha1
kind: Operator
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operators.coreos.com/v2alpha1","kind":"Operator","metadata":{"annotations":{},"name":"etcd-test"},"spec":{"updates":{"catalogSource":{"approval":"Automatic","channel":"singlenamespace-alpha","entrypoint":"etcdoperator.v0.9.4","package":"etcd","ref":{"name":"community-operators","namespace":"openshift-marketplace"}},"type":"CatalogSource"}}}
  creationTimestamp: "2020-06-30T07:45:58Z"
  generation: 1
  managedFields:
...
  name: etcd-test
  resourceVersion: "157367"
  selfLink: /apis/operators.coreos.com/v2alpha1/operators/etcd-test
  uid: b38ec7da-c971-4d42-a1b0-4abceb8b5fa5
spec: {}
status:
  components:
    labelSelector:
      matchExpressions:
      - key: operators.coreos.com/etcd-test
        operator: Exists
    refs:
    - apiVersion: apiextensions.k8s.io/v1
      conditions:
      - lastTransitionTime: "2020-06-30T07:48:00Z"
        message: no conflicts found
        reason: NoConflicts
        status: "True"
        type: NamesAccepted
      - lastTransitionTime: "2020-06-30T07:48:00Z"
        message: the initial names have been accepted
        reason: InitialNamesAccepted
        status: "True"
        type: Established
      kind: CustomResourceDefinition
      name: etcdrestores.etcd.database.coreos.com
...
    - apiVersion: operators.coreos.com/v1alpha1
      conditions:
      - lastTransitionTime: "2020-06-30T07:47:32Z"
        message: all available catalogsources are healthy
        reason: AllCatalogSourcesHealthy
        status: "False"
        type: CatalogSourcesUnhealthy
      kind: Subscription
      name: etcd
      namespace: default
    - apiVersion: operators.coreos.com/v1alpha1
      conditions:
      - lastTransitionTime: "2020-06-30T08:16:05Z"
        lastUpdateTime: "2020-06-30T08:16:05Z"
        status: "True"
        type: Installed
      kind: InstallPlan
      name: install-6c5mr
      namespace: default
    - apiVersion: operators.coreos.com/v1alpha1
      conditions:
      - lastTransitionTime: "2020-06-30T08:16:05Z"
        lastUpdateTime: "2020-06-30T08:16:05Z"
        message: requirements not yet checked
        reason: RequirementsUnknown
        status: ""
        type: ""
      - lastTransitionTime: "2020-06-30T08:16:05Z"
        lastUpdateTime: "2020-06-30T08:16:05Z"
        message: all requirements found, attempting install
        reason: AllRequirementsMet
        status: ""
        type: ""
      - lastTransitionTime: "2020-06-30T08:16:05Z"
        lastUpdateTime: "2020-06-30T08:16:05Z"
        message: waiting for install components to report healthy
        reason: InstallSucceeded
        status: ""
        type: ""
      - lastTransitionTime: "2020-06-30T08:16:05Z"
        lastUpdateTime: "2020-06-30T08:16:06Z"
        message: |
          installing: waiting for deployment etcd-operator to become ready: Waiting for rollout to finish: 1 old replicas are pending termination...
        reason: InstallWaiting
        status: ""
        type: ""
      - lastTransitionTime: "2020-06-30T08:16:15Z"
        lastUpdateTime: "2020-06-30T08:16:15Z"
        message: install strategy completed with no errors
        reason: InstallSucceeded
        status: ""
        type: ""
      kind: ClusterServiceVersion
      name: etcdoperator.v0.9.4
      namespace: default

```

Reply:
The `oc describe operator` command worked for me and yielded the example output I included; is that OK to use?

Reply:
OK. if you want to use "oc describe", it is "oc describe operator etcd-test", not "oc describe etcd-test"

Reply:
Oh whoops 😬 Thank you!

Reply:
the modification is correct. thanks